client/my-sites/github-deployments/components/repositories/create-repository/index.tsx

```diff
@@ -32,37 +29,50 @@ export const CreateRepository = () => {
 
 	const dispatch = useDispatch();
 
-	const { createDeploymentAndRepository, isPending } = useCreateCodeDeploymentAndRepository(
-		siteId as number,
-		{
-			onSuccess: () => {
-				goToDeployments();
-				dispatch( successNotice( __( 'Deployment created.' ), noticeOptions ) );
-			},
-			onError: ( error ) => {
-				dispatch(
-					errorNotice(
-						// translators: "reason" is why connecting the branch failed.
-						sprintf( __( 'Failed to create repository: %(reason)s' ), { reason: error.message } ),
-						{
-							...noticeOptions,
-						}
-					)
-				);
-			},
-			onSettled: ( _, error ) => {
-				dispatch(
-					recordTracksEvent( 'calypso_hosting_github_create_repository_success', {
-						connected: ! error,
-					} )
-				);
-			},
-		}
-	);
+	const { createRepository, isPending } = useCreateRepository( {
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why creating the repository failed.
+					sprintf( __( 'Failed to create repository: %(reason)s' ), { reason: error.message } ),
+					{
+						...noticeOptions,
+					}
+				)
+			);
+		},
+	} );
+	const { createDeployment, isPending: isDeploying } = useCreateCodeDeployment( siteId as number, {
+		onSuccess: () => {
+			goToDeployments();
+			dispatch( successNotice( __( 'Repository connected' ), noticeOptions ) );
+		},
+		onError: ( error ) => {
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why creating the repository failed.
+					sprintf( __( 'Failed to connect repository: %(reason)s' ), {
+						reason: error.message,
+					} ),
+					{
+						...noticeOptions,
+					}
+				)
+			);
+		},
+	} );
 
-	function handleCreateRepository( args: MutationVariables ) {
-		createDeploymentAndRepository( args );
-	}
+	const handleCreateRepository = ( args: OnRepositoryCreatedParams ) => {
+		createRepository( args ).then( ( response ) => {
+			createDeployment( {
```

Comment:
A nitpick, but: since we're already using react query hooks, does it make sense to call this function inside the `onSuccess` hook of the create repository call?